numba/tests/test_parfors.py

```diff
@@ -3981,6 +3983,107 @@ def test_issue_due_to_max_label(self):
         self.assertIn("TEST PASSED", out.decode())
 
 
+class TestParforChunksizing(TestCase):
+    """
+    Tests miscellaneous parts of ParallelAccelerator use.
+    """
+    _numba_parallel_test_ = False
+
+    @skip_parfors_unsupported
+    def test_python_parallel_chunksize_basic(self):
+        # Test basic chunksize operations outside njit.
+        self.assertEqual(get_parallel_chunksize(), 0)
+        set_parallel_chunksize(8)
+        self.assertEqual(get_parallel_chunksize(), 8)
+        set_parallel_chunksize(0)
+        self.assertEqual(get_parallel_chunksize(), 0)
+
+    @skip_parfors_unsupported
+    def test_python_with_chunksize(self):
+        # Test "with parallel_chunksize" outside njit.
+        self.assertEqual(get_parallel_chunksize(), 0)
+        with parallel_chunksize(8):
+            self.assertEqual(get_parallel_chunksize(), 8)
+        self.assertEqual(get_parallel_chunksize(), 0)
+
+    @skip_parfors_unsupported
+    def test_njit_parallel_chunksize_basic(self):
+        # Test basic chunksize operations inside njit.
+        @njit
+        def get_cs():
+            return get_parallel_chunksize()
+
+        @njit
+        def set_cs(x):
+            return set_parallel_chunksize(x)
+
+        self.assertEqual(get_cs(), 0)
+        set_cs(8)
+        self.assertEqual(get_cs(), 8)
+        set_cs(0)
+        self.assertEqual(get_cs(), 0)
+
+    @skip_parfors_unsupported
+    def test_njit_with_chunksize(self):
+        # Test "with parallel_chunksize" inside njit.
+        @njit
+        def test_impl(x):
+            cs1 = get_parallel_chunksize()
+            with parallel_chunksize(8):
+                cs2 = get_parallel_chunksize()
+            cs3 = get_parallel_chunksize()
+            return cs1, cs2, cs3
+
+        cs1, cs2, cs3 = test_impl(8)
+
+        self.assertEqual(cs1, 0)
+        self.assertEqual(cs2, 8)
+        self.assertEqual(cs3, 0)
+
+    @skip_parfors_unsupported
+    def test_all_iterations_reset_chunksize(self):
+        """ Test that all the iterations get run if you set the
+            chunksize.  Also check that the chunksize that each
+            worker thread sees has been reset to 0. """
+
+        @njit(parallel=True)
+        def test_impl(cs):
+            n = 1000
```

Comment:
Perhaps check a number of different array sizes and chunk sizes. 1000/8=125 so it's a relatively safe case?